src/methods/join.js

```diff
@@ -700,9 +726,9 @@ export class JoinImplementation {
 
   composeGenerators(realm: Realm, generator1: Generator, generator2: Generator): Generator {
     let result = new Generator(realm, "composed");
-    if (!generator1.empty() || !generator2.empty()) {
-      result.composeGenerators(generator1, generator2);
-    }
+    // We copy the entries here because actually composing the generators breaks the serializer
```

Comment:
Why?

Reply:
It seems like a bug or short coming in the serializer. I do not quite understand why.

Reply:
That is concerning; maybe hiding something. After landing this, can you file an issue with steps to reproduce?
(E.g. the old code to plug in, and which test case fails.)